### PR TITLE
Apply `black` to `astropy.cosmology`

### PR DESCRIPTION
**Pull request by WilliamJamieson**
### Description

Apply `black` to `astropy.cosmology` according to the process laid out by [APE 20](https://github.com/astropy/astropy-APEs/blob/main/APE20.rst).

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by nstarman** [comment]
@WilliamJamieson, `black` is failing in `pre-commit`.

**Comment by WilliamJamieson** [comment]
> @WilliamJamieson, `black` is failing in `pre-commit`.

It's supposed to be. Per APE 20 I am supposed to run with the --skip-string-normalization flag. That way the initial review is not overwhelmed by string normalization.

**Comment by nstarman** [comment]
Oh, then shouldn't that should be added to the pre-commit config?

**Comment by WilliamJamieson** [comment]
@nstarman I have temporarily suppressed string normalization in the pre-commit config and removed the pre-commit bot's commit which normalizes the strings.

I also ran `black --skip-string-normalization --preview .` on cosmology as that cleans up quite a bit of things. It is a separate commit if you want to see what exactly it altered.

File: astropy/cosmology/core.py [hunk 1]
```diff
@@ -328,9 +332,12 @@ def __equiv__(self, other: Any, /) -> bool:
...
         # no extra parameters (latter part should never happen b/c same class)
-        params_eq = (set(self.__all_parameters__) == set(other.__all_parameters__)
-                     and all(np.all(getattr(self, k) == getattr(other, k))
-                             for k in self.__all_parameters__))
+        params_eq = set(self.__all_parameters__) == set(
+            other.__all_parameters__
+        ) and all(
+            np.all(getattr(self, k) == getattr(other, k))
+            for k in self.__all_parameters__
+        )
```
Comment (nstarman) [inline astropy/cosmology/core.py:340]:
```suggestion
        params_eq = (
            (set(self.__all_parameters__) == set(other.__all_parameters__)
            and all(
                np.all(getattr(self, k) == getattr(other, k)) for k in self.__all_parameters__
            )
        )
```

Does this pass black?

File: astropy/cosmology/flrw/base.py [hunk 2]
```diff
@@ -1096,7 +1146,9 @@ def _integral_comoving_distance_z1z2(self, z1, z2):
         d : `~astropy.units.Quantity` ['length']
             Comoving distance in Mpc between each input redshift.
         """
-        return self._hubble_distance * self._integral_comoving_distance_z1z2_scalar(z1, z2)
+        return self._hubble_distance * self._integral_comoving_distance_z1z2_scalar(
+            z1, z2
+        )
```
Comment (nstarman) [inline astropy/cosmology/flrw/base.py:1151]: Don't love this.

File: astropy/cosmology/io/utils.py [hunk 3]
```diff
@@ -62,11 +66,26 @@ def convert_parameter_to_model_parameter(parameter, value, meta=None):
...
+            'setter',
+            'fixed',
+            'tied',
+            'min',
+            'max',
+            'bounds',
+            'prior',
+            'posterior',
+        )
+    }
```
Comment (nstarman) [inline astropy/cosmology/io/utils.py:84]:
```suggestion
    attrs = ('getter', 'setter', 'fixed', 'tied', 'min', 'max', 'bounds', 'prior', 'posterior')
    extra = {k: v for k, v in (meta or {}).items() if k in attrs}
```

File: astropy/cosmology/io/tests/test_json.py [hunk 4]
```diff
@@ -127,9 +130,13 @@ def test_readwrite_json_subclass_partial_info(self, cosmo_cls, cosmo, read,
         # partial information
         with open(fp) as file:
             L = file.readlines()[0]
-        L = L[: L.index('"cosmology":')] + L[L.index(", ") + 2 :]  # remove cosmology  # noqa: #203
+        L = (
```
Comment (nstarman) [inline astropy/cosmology/io/tests/test_json.py:133]: A little gross, but fine for now.

**Review by nstarman — APPROVED** [review]
Thanks @WilliamJamieson . Mostly LGTM. I flagged a couple small things.

File: astropy/cosmology/core.py (hunk 1 above)
Comment (WilliamJamieson) [reply astropy/cosmology/core.py:340]: I think there is a missing `)` in there and when I add that, black corrects it back to the original.

However, if one looks one line below (line 341 in this case) we find we just return `params_eq` so if I replace `param_eq` with `return` and remove the original `return` we get:

```python
        return set(self.__all_parameters__) == set(other.__all_parameters__) and all(
            np.all(getattr(self, k) == getattr(other, k))
            for k in self.__all_parameters__
        )
```
after black formats the file. Is this better?

File: astropy/cosmology/flrw/base.py (hunk 2 above)
Comment (WilliamJamieson) [reply astropy/cosmology/flrw/base.py:1151]: I could add a `# fmt: skip` and make it one line?

File: astropy/cosmology/io/utils.py (hunk 3 above)
Comment (WilliamJamieson) [reply astropy/cosmology/io/utils.py:84]: black reformats your suggestion to:
```python
    attrs = (
        'getter',
        'setter',
        'fixed',
        'tied',
        'min',
        'max',
        'bounds',
        'prior',
        'posterior',
    )
    extra = {k: v for k, v in (meta or {}).items() if k in attrs}
```

Do you still want to keep it?

File: astropy/cosmology/io/tests/test_json.py (hunk 4 above)
Comment (WilliamJamieson) [reply astropy/cosmology/io/tests/test_json.py:133]: Again we can `# fmt: skip` if you would like.

File: astropy/cosmology/core.py (hunk 1 above)
Comment (nstarman) [reply astropy/cosmology/core.py]: This does look better!

File: astropy/cosmology/flrw/base.py (hunk 2 above)
Comment (nstarman) [reply astropy/cosmology/flrw/base.py:1151]: I think so. At least for now.

File: astropy/cosmology/io/utils.py (hunk 3 above)
Comment (nstarman) [reply astropy/cosmology/io/utils.py:84]: Yeah. It separates the logic better.

File: astropy/cosmology/io/tests/test_json.py (hunk 4 above)
Comment (nstarman) [reply astropy/cosmology/io/tests/test_json.py:133]: No, I'd rather rewrite it to be more explicit about the string magic. An endeavor for a future pull request.

**Review by nstarman — APPROVED** [review]
Took another quick look. Thanks @WilliamJamieson !

**Comment by WilliamJamieson** [comment]
@nstarman if this PR is ready for string normalization, please move it to the "Ready for string normalization" column in https://github.com/orgs/astropy/projects/10.

**Comment by WilliamJamieson** [comment]
> @nstarman if this PR is ready for string normalization, please move it to the "Ready for string normalization" column in [github.com/orgs/astropy/projects/10](https://github.com/orgs/astropy/projects/10).

Thanks for moving this forward. I have now normalized the strings. If things look good please move this forward to the "Ready for Squash" column.

File: astropy/cosmology/__init__.py [hunk 5]
```diff
@@ -19,10 +19,15 @@
...
-           + funcs.__all__ + parameter.__all__ + utils.__all__)  # utils
+__all__ = (
+    core.__all__
+    + flrw.__all__  # cosmology classes
+    + realizations.__all__  # instances thereof
+    + ["units"]
+    + funcs.__all__
+    + parameter.__all__
+    + utils.__all__
+)  # utils
```
Comment (nstarman) [inline astropy/cosmology/__init__.py:30]:
```suggestion
    # utils
    + funcs.__all__
    + parameter.__all__
    + utils.__all__
)
```

**Review by nstarman — COMMENTED** [review]
One small correction that slipped through the cracks.

**Pull request merged**